codalab/lib/formatting.py

```diff
@@ -22,6 +22,8 @@ def contents_str(input_string):
     if input_string is None:
         return ''
 
+    if isinstance(input_string, unicode):
+        return input_string
     try:
         input_string.decode('utf-8')
```

Comment:
Actually I believe this is correct, since arbitrary streams of bytes should be stored as strings in Python2 (there is a new `bytes` type in Python3 to disambiguate). `unicode` type is only for Unicode data.

I can clean up the logic here and add comments to clarify.